src/main/java/Parser.java

```diff
@@ -0,0 +1,129 @@
+                        System.out.println(current.getStatusIcon() + " " + current.getDescription());
+                    } catch (IndexOutOfBoundsException e) {
+                        throw new DukeException("delete2");
+                    } catch (InputMismatchException e) {
+                        throw new DukeException(command);
+                    }
+                } else if (command.startsWith("todo")) {
+                    String desc = command.split(" ")[1];
+                    if (desc.equals("")) {
+                        throw new DukeException(command);
+                    }
+                    Todo todo = new Todo(desc);
+
+                    store.save(storage, todo);
+                    int size = storage.size();
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + todo);
+                    System.out.println("Now you have " + size + " tasks in the list.");
+                } else if (command.startsWith("deadline")) {
+                    String desc = command.split("/by ")[1];
+                    if (desc.equals("")) {
+                        throw new DukeException(command);
+                    }
+                    String[] string = command.split("/by ");
+                    if (string.length < 2) {
+                        throw new DukeException(command);
+                    }
+                    Deadline deadline = new Deadline(string[0], string[1]);
+                    store.save(storage, deadline);
+                    int size = storage.size();
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + deadline);
+                    System.out.println("Now you have " + size + " tasks in the list.");
+                } else if (command.startsWith("event")) {
+                    String desc = command.split("/at ")[1];
+                    if (desc.equals("")) {
+                        throw new DukeException(command);
+                    }
+                    String[] string = command.split("/at ");
+                    if (string.length < 2) {
+                        throw new DukeException(command);
+                    }
+                    Events event = new Events(string[0], string[1]);
+                    store.save(storage, event);
+                    int size = storage.size();
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + event);
+                    System.out.println("Now you have " + size + " tasks in the list.");
+                } else if (command.equals("bye")) {
+                    String bye = "Bye. Hope to see you again soon!";
+                    System.out.println(bye);
+                    scan1.close();
+                    break;
+                } else if (command.startsWith("delete")) {
```

Comment:
When the code gets long, perhaps it would be a good idea to add single line comments.